### PR TITLE
Preflight commands

### PR DESCRIPTION
**Pull request by ashwathishiva**
`qliksense preflight createRole`
`qliksense preflight createRoleBinding`
`qliksense preflight createServiceAccount`
`qliksense preflight createRB`
`qliksense preflight mongo --url=<url>`
`qliksense preflight mongo` 
`qliksense preflight all --mongodb-url=<url>`

TODO:
- get mongodb check working for real mongodbUrl instead of local mongo instance
- supply command as a shell script instead of exec-ing into the container.
- unit tests

### CODE REVIEW

**Comment by ffoysal** [comment]
```
bin/qliksense preflight createRB
Preflight createRB check
------------------------
Preflight createRole check:
Unable to retrieve manifests from executing kustomize: Error creating new loader with git: uri looks like abs path: /var/folders/tg/2ftr8gqs6yl4_dryp980yxwh0000gp/T/750087972/repo/Users/mqb/.qliksense/contexts/qlik-default/qlik-k8s/v0.0.6/manifests/docker-desktop, dir: evalsymlink failure on '/var/folders/tg/2ftr8gqs6yl4_dryp980yxwh0000gp/T/750087972/repo/Users/mqb/.qliksense/contexts/qlik-default/qlik-k8s/v0.0.6/manifests/docker-desktop' : lstat /private/var/folders/tg/2ftr8gqs6yl4_dryp980yxwh0000gp/T/750087972/repo/Users: no such file or directory, get: invalid source string: /var/folders/tg/2ftr8gqs6yl4_dryp980yxwh0000gp/T/750087972/repo/Users/mqb/.qliksense/contexts/qlik-default/qlik-k8s/v0.0.6/manifests/docker-desktop
Preflight createRole check: FAILED
Completed preflight create-role check

Preflight createRoleBinding check:
Unable to retrieve manifests from executing kustomize: Error creating new loader with git: uri looks like abs path: /var/folders/tg/2ftr8gqs6yl4_dryp980yxwh0000gp/T/073005789/repo/Users/mqb/.qliksense/contexts/qlik-default/qlik-k8s/v0.0.6/manifests/docker-desktop, dir: evalsymlink failure on '/var/folders/tg/2ftr8gqs6yl4_dryp980yxwh0000gp/T/073005789/repo/Users/mqb/.qliksense/contexts/qlik-default/qlik-k8s/v0.0.6/manifests/docker-desktop' : lstat /private/var/folders/tg/2ftr8gqs6yl4_dryp980yxwh0000gp/T/073005789/repo/Users: no such file or directory, get: invalid source string: /var/folders/tg/2ftr8gqs6yl4_dryp980yxwh0000gp/T/073005789/repo/Users/mqb/.qliksense/contexts/qlik-default/qlik-k8s/v0.0.6/manifests/docker-desktop
Preflight createRoleBinding check: FAILED
Completed preflight createRoleBinding check

Preflight createServiceAccount check:
Unable to retrieve manifests from executing kustomize: Error creating new loader with git: uri looks like abs path: /var/folders/tg/2ftr8gqs6yl4_dryp980yxwh0000gp/T/505991050/repo/Users/mqb/.qliksense/contexts/qlik-default/qlik-k8s/v0.0.6/manifests/docker-desktop, dir: evalsymlink failure on '/var/folders/tg/2ftr8gqs6yl4_dryp980yxwh0000gp/T/505991050/repo/Users/mqb/.qliksense/contexts/qlik-default/qlik-k8s/v0.0.6/manifests/docker-desktop' : lstat /private/var/folders/tg/2ftr8gqs6yl4_dryp980yxwh0000gp/T/505991050/repo/Users: no such file or directory, get: invalid source string: /var/folders/tg/2ftr8gqs6yl4_dryp980yxwh0000gp/T/505991050/repo/Users/mqb/.qliksense/contexts/qlik-default/qlik-k8s/v0.0.6/manifests/docker-desktop
Preflight createServiceAccount check: FAILED
Completed preflight createServiceAccount check
```

**Comment by ffoysal** [comment]
can we avoid camel case here
```
qliksense preflight createRole
qliksense preflight createRoleBinding
qliksense preflight createServiceAccount
qliksense preflight createRB
```

all lower case seems standard
`qliksense perflight role`
`qliksense preflight rolebinding`
`qliksense preflight serviceaccount`

moreover thinking to combine all in  one command like this
`qliksense preflight qseok authorization` or `qliksense preflight authcheck` or any better name
since we are basically checking all roles and rolebinding, serviceaccount dynamically.

File: pkg/preflight/role_check.go [hunk 1]
```diff
@@ -1,12 +1,134 @@
...
+	fmt.Println("Completed preflight serviceaccount check")
+	return nil
+}
+func (qp *QliksensePreflight) checkCreateEntity(namespace, entityToTest string) error {
+	qConfig := qapi.NewQConfig(qp.Q.QliksenseHome)
+	var currentCR *qapi.QliksenseCR
+	var tempDownloadedDir string
+	var resultYamlString []byte
+	var err error
+	qConfig.SetNamespace(namespace)
```
Comment (ffoysal) [inline pkg/preflight/role_check.go]: we dont need it

File: pkg/preflight/role_check.go [hunk 2]
```diff
@@ -1,12 +1,134 @@
...
+	currentCR, err = qConfig.GetCurrentCR()
+	if err != nil {
+		fmt.Printf("Unable to retrieve current CR: %v\n", err)
+		return err
+	}
+
+	if tempDownloadedDir, err = qliksense.DownloadFromGitRepoToTmpDir(qliksense.QLIK_GIT_REPO, "master"); err != nil {
```
Comment (ffoysal) [inline pkg/preflight/role_check.go]: `if currentCR.IsRepoExist(){ dont need to download to temp dir }`

File: cmd/qliksense/preflight.go [hunk 3]
```diff
@@ -186,35 +186,144 @@ func pfPodCheckCmd(q *qliksense.Qliksense) *cobra.Command {
...
+			}
+			return nil
+		},
+	}
+	return preflightServiceAccountCmd
+}
+
+func pfCreateRBCheckCmd(q *qliksense.Qliksense) *cobra.Command {
+	var preflightCreateRBCmd = &cobra.Command{
+		Use:     "RB",
```
Comment (ffoysal) [inline cmd/qliksense/preflight.go]: can we change it to `access` or `authcheck`, something like this

Comment (ashwathishiva) [reply cmd/qliksense/preflight.go]: done, renamed to `authcheck`.

**Review by ffoysal — APPROVED** [review]

**Pull request merged**